### PR TITLE
change createFiles to Publish-Subscriber pattern

### PR DESCRIPTION
**Pull request by a5x55555**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1032

### CODE REVIEW

File: src/ProjectService.Broker/Publishes/Publish.cs [hunk 1]
```diff
@@ -5,29 +5,45 @@
...
+using LT.DigitalOffice.Models.Broker.Models.File;
+using LT.DigitalOffice.Models.Broker.Publishing.Subscriber.File;
 using LT.DigitalOffice.ProjectService.Broker.Publishes.Interfaces;
 using MassTransit;
+using Microsoft.AspNetCore.Http;
+using LT.DigitalOffice.Kernel.Extensions;
 
 namespace LT.DigitalOffice.ProjectService.Broker.Publishes
 {
+
```
Comment (VLiamin) [inline src/ProjectService.Broker/Publishes/Publish.cs:17]: empty line

File: src/ProjectService.Broker/Publishes/Publish.cs [hunk 2]
```diff
@@ -5,29 +5,45 @@
...
-      IBus bus)
+      IBus bus,
+      IHttpContextAccessor httpContextAccessor)
     {
       _bus = bus;
+      _httpContextAccessor = httpContextAccessor;
     }
 
     public async Task CreateDepartmentEntityAsync(Guid departmentId, Guid createdBy, Guid projectId)
+
```
Comment (VLiamin) [inline src/ProjectService.Broker/Publishes/Publish.cs:32]: empty line

Comment (VLiamin) [inline src/ProjectService.Broker/Publishes/Publish.cs:32]: empty line

File: src/ProjectService.Broker/Publishes/Publish.cs (hunk 1 above)
Comment (VLiamin) [inline src/ProjectService.Broker/Publishes/Publish.cs:17]: empty line

File: src/ProjectService.Broker/Publishes/Publish.cs [hunk 3]
```diff
@@ -5,29 +5,45 @@
...
       await _bus.Publish<ICreateDepartmentEntityPublish>(ICreateDepartmentEntityPublish.CreateObj(
         departmentId: departmentId,
         createdBy: createdBy,
         projectId: projectId));
     }
 
+    public async Task CreateFilesAsync(List<FileData> files)
+    {
+      await _bus.Publish<ICreateFilesPublish>(ICreateFilesPublish.CreateObj(
+      files: files,
```
Comment (VLiamin) [inline src/ProjectService.Broker/Publishes/Publish.cs:43]: spaces

File: src/ProjectService.Business/Commands/File/CreateFilesCommand.cs [hunk 4]
```diff
@@ -4,61 +4,51 @@
...
+using LT.DigitalOffice.ProjectService.Broker.Publishes.Interfaces;
 using LT.DigitalOffice.ProjectService.Data.Interfaces;
 using LT.DigitalOffice.ProjectService.Mappers.Db.Interfaces;
 using LT.DigitalOffice.ProjectService.Mappers.Models.Interfaces;
 using LT.DigitalOffice.ProjectService.Models.Dto.Models;
 using LT.DigitalOffice.ProjectService.Models.Dto.Requests;
-using MassTransit;
 using Microsoft.AspNetCore.Http;
-using Microsoft.Extensions.Logging;
+
```
Comment (VLiamin) [inline src/ProjectService.Business/Commands/File/CreateFilesCommand.cs:19]: empty line

File: src/ProjectService.Models.Dto/Requests/CreateFilesRequest.cs [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using LT.DigitalOffice.ProjectService.Models.Dto.Models;
 
 namespace LT.DigitalOffice.ProjectService.Models.Dto.Requests
 {
   public record CreateFilesRequest
   {
     public Guid ProjectId { get; set; }
+
```
Comment (VLiamin) [inline src/ProjectService.Models.Dto/Requests/CreateFilesRequest.cs:11]: In FileInfo.cs you do not make empty line.

File: src/ProjectService.Models.Dto/Requests/CreateProjectRequest.cs [hunk 6]
```diff
@@ -10,6 +10,8 @@ public record CreateProjectRequest
   {
     [Required]
     public string Name { get; set; }
+
```
Comment (VLiamin) [inline src/ProjectService.Models.Dto/Requests/CreateProjectRequest.cs:13]: In FileInfo.cs you do not make empty line.

**Review by VLiamin — APPROVED** [review]

File: src/ProjectService/appsettings.json [hunk 7]
```diff
@@ -73,7 +73,6 @@
     "CreateImagesEndpoint": "ImageService_CreateImages",
 
     //FileService
```
Comment (MariyaSin) [inline src/ProjectService/appsettings.json:75]: remove

File: src/ProjectService.Broker/Publishes/Publish.cs [hunk 8]
```diff
@@ -6,19 +6,26 @@
 using LT.DigitalOffice.Models.Broker.Publishing.Subscriber.File;
 using LT.DigitalOffice.Models.Broker.Publishing.Subscriber.Image;
 using LT.DigitalOffice.Models.Broker.Publishing.Subscriber.Time;
+using LT.DigitalOffice.Models.Broker.Models.File;
+using LT.DigitalOffice.Models.Broker.Publishing.Subscriber.File;
```
Comment (ndoriomedov) [inline src/ProjectService.Broker/Publishes/Publish.cs:10]: remove

File: src/ProjectService.Business/Commands/File/CreateFilesCommand.cs [hunk 9]
```diff
@@ -4,61 +4,50 @@
...
-    private readonly ILogger<CreateFilesCommand> _logger;
     private readonly IAccessValidator _accessValidator;
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly IProjectUserRepository _userRepository;
     private readonly IResponseCreator _responseCreator;
     private readonly IFileDataMapper _fileDataMapper;
-    private readonly IFileService _fileService;
-
+    private readonly IPublish _publish;
+
```
Comment (ndoriomedov) [inline src/ProjectService.Business/Commands/File/CreateFilesCommand.cs:32]: remove spaces

File: src/ProjectService.Broker/Publishes/Publish.cs (hunk 8 above)
Comment (MariyaSin) [inline src/ProjectService.Broker/Publishes/Publish.cs:10]: remove the using

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**